### DISCUSSION TITLE
Gltfio material samplerExternal

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [google/filament#3784] [Q&A] [answered]
Hi,

Can you just confirm that since there are only SAMPLER_2D generated by the [gltfio MaterialGenerator](https://github.com/google/filament/blob/main/libs/gltfio/src/MaterialGenerator.cpp) even for the 
```
// Transmission Map
{ type : int, name : transmissionIndex },
{ type : float, name : transmissionFactor },
{ type : sampler2d, name : transmissionMap },
{ type : mat3, name : transmissionUvMatrix }
```
there is no way I can use the default glTF material to apply an external Stream parameter without swiping the whole material to a self made .filamat?

Thanks

**Comment by prideout** [comment, marked as answer]
Correct, you'll need to rebind your primitive to an alternative material if the built-in glTF material doesn't suit your needs.

If your application is written in C++, one other possibility is subclassing MaterialProvider.

**Reply by ThomasGorisse** [reply]
Thanks

**Comment by ThomasGorisse** [comment]
For anyone looking for a sample, here is an implementation using MaterialBuilder in a Sceneform context :
[Video Texture Sample](https://www.github.com/ThomasGorisse/sceneform-android-sdk/tree/master/samples%2Fvideo-texture)
